Sources/Sentry/SentryProfiler.mm

```diff
@@ -264,6 +266,136 @@
     return result;
 }
 
+NSDictionary<NSString *, id> *
+serializedProfileData(NSDictionary<NSString *, id> *profileData, SentryTransaction *transaction,
+    SentryId *profileID, NSString *truncationReason, NSString *environment, NSString *release,
+    NSDictionary<NSString *, id> *serializedMetrics, NSArray<SentryDebugMeta *> *debugMeta)
+{
+    NSArray<SentrySample *> *samplesCopy;
+    NSArray<NSArray<NSNumber *> *> *stacksCopy;
+    NSArray<NSDictionary<NSString *, id> *> *framesCopy;
+    NSDictionary<NSString *, NSDictionary *> *threadMetadataCopy;
+    NSDictionary<NSString *, NSDictionary *> *queueMetadataCopy;
+    {
+        std::lock_guard<std::mutex> d(_gDataStructureLock);
+        samplesCopy = [profileData[@"profile"][@"samples"] copy];
```

Comment:
Definitely. @indragiek and I have discussed a memory footprint optimization of using a ring-buffer, which will be a good thing to keep in mind for continuous profiling, I think it would integrate well with the needs to stream chunks of profiling data to the server.